File: implementation/scipamato/core/pubmed-api/src/intTest/java/ch/difty/scipamato/core/pubmed/PubmedIntegrationTest.java
package ch.difty.scipamato.core.pubmed;

import static ch.difty.scipamato.common.TestUtilsKt.readFileAsString;
import static org.assertj.core.api.Assertions.assertThat;

import java.io.IOException;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import ch.difty.scipamato.core.pubmed.api.PubmedArticleSet;

@SpringBootTest
@SuppressWarnings("SameParameterValue")
abstract class PubmedIntegrationTest {

    @Autowired
    private PubmedXmlService service;

    PubmedArticleSet getPubmedArticleSet(String fileName) throws IOException {
        String xml = readFileAsString(fileName);
        assertThat(xml).isNotNull();
        return service.unmarshal(xml);
    }

    List<PubmedArticleFacade> getPubmedArticles(String fileName) throws IOException {
        String xml = readFileAsString(fileName);
        assertThat(xml).isNotNull();
        return service.extractArticlesFrom(xml);
    }

}

File: core/core-pubmed-api/src/integration-test/kotlin/ch/difty/scipamato/core/pubmed/PubmedIntegrationTest.kt
package ch.difty.scipamato.core.pubmed

import ch.difty.scipamato.common.readFileAsString
import ch.difty.scipamato.core.pubmed.api.PubmedArticleSet
import org.amshove.kluent.shouldNotBeNull
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@Suppress("SpellCheckingInspection")
@SpringBootTest
internal abstract class PubmedIntegrationTest {

    @Autowired
    private lateinit var service: PubmedXmlService

    fun getPubmedArticleSet(fileName: String): PubmedArticleSet? {
        val xml = readFileAsString(fileName)
        xml.shouldNotBeNull()
        return service.unmarshal(xml)
    }

    fun getPubmedArticles(fileName: String): List<PubmedArticleFacade> {
        val xml = readFileAsString(fileName)
        xml.shouldNotBeNull()
        return service.extractArticlesFrom(xml)
    }
}
